src/ray/raylet/node_manager.cc

```diff
@@ -1938,6 +1938,29 @@ void NodeManager::HandleReturnWorker(const rpc::ReturnWorkerRequest &request,
   send_reply_callback(status, nullptr, nullptr);
 }
 
+void NodeManager::HandleReleaseUnusedWorkers(
+    const rpc::ReleaseUnusedWorkersRequest &request,
+    rpc::ReleaseUnusedWorkersReply *reply, rpc::SendReplyCallback send_reply_callback) {
+  std::unordered_set<WorkerID> used_worker_ids;
+  for (int index = 0; index < request.worker_id_list_size(); ++index) {
+    auto worker_id = WorkerID::FromBinary(request.worker_id_list(index));
+    used_worker_ids.emplace(worker_id);
+  }
+
+  std::vector<WorkerID> unused_worker_ids;
+  for (auto &iter : leased_workers_) {
+    if (!used_worker_ids.count(iter.first)) {
+      unused_worker_ids.emplace_back(iter.first);
+    }
+  }
+
+  for (auto &iter : unused_worker_ids) {
+    leased_workers_.erase(iter);
```

Comment:
Isn't it erasing workers that are used by other core workers? How does it guarantee `leased_workers` only contain workers that have been used by GCS server?

Reply:
> Isn't it erasing workers that are used by other core workers? How does it guarantee `leased_workers` only contain workers that have been used by GCS server?

We want to delete the workers leased but not used by GCS. The worker of common task may be deleted by mistake. I'll see how to solve it. Thank you